### DISCUSSION TITLE
External workflows definition

### DISCUSSION THREAD

**Discussion by aprzybys** [community/community#9583] [Actions]
My team contributes to external (open source) repositories. To do so, we need to mirror the repository (in case of GH: create custom action updating master/main branch, since there is no such an option) and then work on branches in such a mirror (diverged from "mirrored" repo main branch). The problem is, we cannot configure our (custom) workflows, as we obviously cannot modify "mirrored" repo main branch. We have nowhere to put them (they won't run from branches other than main branch).

In short: we miss a feature analogous to https://docs.gitlab.com/ee/ci/pipelines/settings.html#specify-a-custom-cicd-configuration-file with possibility to pass a path to files (workflows) in another project.

**Comment by jsoref** [comment]
There are ways to cheat.

One is to create a default branch in your repo with your workflows, and then create PRs (that you *never* close) from branches that are synced to the upstream branches.

I'm actively testing a workflow for an upstream and have open PRs that track PRs to the upstream, so each time a push is made to the source branch for the upstream PR, I get a workflow bump to my fork.

You could also rely on `on: comment` or something and have your workflow check out the commits you care about and do what you need. 

You could rely on on `on: schedule` and using a bot token, have it create comments (or pull requests) to do whatever you need done.

**Comment by ethomson** [comment]
Thanks for the feedback @aprzybys.  We don't have any immediate plans for this, but I appreciate your use-case.  I'll put this on our backlog.
